### ISSUE TITLE
Add "TabGroups Menu" capabilities

### ISSUE THREAD

**Issue by 4udart** [Quicksaver/Tab-Groups#5]
Hi! Thank you for addon! 

Can you add the functional of TabGroups Menu — https://addons.mozilla.org/en-US/firefox/addon/tabgroups-menu/ ? With the drop down list of tabs in groups. 

Thanks.

**Comment by poke** [comment]
Check out [Simplified Tab Groups](https://addons.mozilla.org/de/firefox/addon/tab-groups/).

Also see #2.

**Comment by the8472** [comment]
As a matter of keeping things modular, wouldn't it be better to ask them to simply interface with this addon?

**Comment by Quicksaver** [comment]
> As a matter of keeping things modular, wouldn't it be better to ask them to simply interface with this addon?

I'm not sure of what you mean. Either way, at least Simplified Tab Groups is sort of doing that already (and I'm almost sure TabGroups Menu does too), it uses the exact same data as this add-on. So whatever groups you create in one add-on will appear in the other. (But you should definitely read #2 as @poke mentioned because the latest version of Simplified Tab Groups on AMO is not fully compatible with this yet.)

**Comment by Quicksaver** [comment]
Either way, I won't be adding this feature to the add-on any time soon at least, because [Simplified Tab Groups](https://addons.mozilla.org/de/firefox/addon/tab-groups/)'s next version should be enough for this. But eventually if that stops being the case, maybe I will. :)

**Comment by B00ze64** [comment]
@Quicksaver Hi Luis. Thanks for working on this addOn. Regarding Simplified Tab Group -> I don't like the way the Simplified Tab Group menu is constructed. Instead of a menu with Groups which are SubMenus, it uses, how to explain it, "openable" menu items (you see what I mean? There's a little arrow to Open/Close groups, but the entries appear in the main menu, not as a SubMenu) and while this is all cute and dandy, it is far slower and painful to use than SubMenus, IMHO it works fine on touch-based devices, but it sucks with a mouse. I'm not saying that you add TabGroups Menu's (and button) functionality to your addOn, just make sure they work nicely together. Thanks. Best Regards.

**Comment by aletaziar** [comment]
I second that this as a very useful extension. Unfortunate that this isn't going to be integrated, but maybe after Panorama is removed from Firefox you can change your button ID (tabGroups-tabview-button) to the old one (tabview-button)? Some extensions will start working again if you do that. TabGroups Menu works if you edit the bootstrap.js in the xpi to use your button ID instead of the default.

**Comment by mikhoul** [comment]
Thanks for the tip @aletaziar I will download TGM right now and edit the button ID. :kissing_closed_eyes:

@Quicksaver What do you think of the suggestion of renaming the button ID after the deprecation of panorama ?

**Comment by Quicksaver** [comment]
As a general rule I avoid renaming stuff, especially because that would make it more ambiguous (less specific identifier); for instance, who's to say someone else won't do the same in their add-on. Toolbar buttons are especially tricky because there's also the overhead of dealing with the customization system in Firefox keeping track of them; I'm not saying it's impossible or even overly hard, just saying it's not *just* changing the id. ;)

Also, since the other day I've seen a lot of support towards menu-like group choosers. While I haven't decided anything yet, who knows, maybe I'll add something like this to the add-on. :)

BTW I couldn't help but notice, you jumped to "hacking TGM" in such a short step, but has anyone tried to contact TGM's developers about this? I'm sure they'd love to add support for Tab Groups in TGM, and I haven't seen any evidence that they're not active anymore; the last update wasn't THAT long ago, and their github account shows some activity from a couple of months back.

**Comment by mikhoul** [comment]
I tried many times over the last months to reach the dev by various channel and no answer at all from him sadly :disappointed:

I really don't know if he care about the addon or Firefox anymore... the only thing I see is lot of devs jumping the boat for supporting addon after the XUL deprecation announcement some don't answer no more to request and some others just remove their addon from AMO.

I save all my XPI files since few months cause lot of addon were removed from AMO by devs that don't want no longer be involved with Firefox. :(

So I'm happy to see that you resurrect Panorama,  I jumped so fast cause I've see lot of users that were using TGM with panorama that try it with your addon but since thee ID confict it was not working... so at least there is a fix they could use :laughing:

**Comment by Quicksaver** [comment]
Ah, I understand. ;) That changes things a bit. It might not happen right away, but I'll see about adding a similar feature to the add-on then, or at least try to make it compatible with TGM on my end.

**Comment by Quicksaver** [comment]
Turns out I can't make TabGroups Menu compatible from within the add-on, at least not without replicating the entirety of its code...

Guess it's decided then, a menu-like function for the add-on is in order. It's not the first item on my list though, so it won't happen right away.

**Comment by mikhoul** [comment]
@Quicksaver   Cool :grinning:  since TGM seem to have a conflict with Adblock Plus, after the first install (restartless addon) it does not initialize on next restart... so you have to drop again the XPI file to see the menu.

**Comment by Quicksaver** [comment]
Guess that TGN's dev is still active a bit at least. Version [0.9.7](https://addons.mozilla.org/firefox/addon/tabgroups-menu/) adds compatibility with this add-on. :)

**Comment by mikhoul** [comment]
@Quicksaver  Sadly the TGM Dev seem to not really care about users, he updated BUT it doesn't fix the incompatibility problem with ABP, so for me it's unusable :disappointed:  

Don't work with ABP: The menu disappear as soon you restart Firefox.

If someone can't fix the XPI file I will be happy to use it even if it's not signed ! :wink:

**Comment by Quicksaver** [comment]
I will still add this feature to the add-on of course, I just thought I'd mention that since I noticed TGM got updated. And it's probably not that they don't care, but that they don't have the availability to work on it anymore, which I completely understand. ;) (of course, I could be wrong and you could be right, but meh, I like to think positive.)

**Comment by mikhoul** [comment]
I'd like to be positive too but the reality is there the dev don't answer  since many months to numerous request from users including myself.

Some dev just said they don't have no more time or it's no longer a priority in their life, it took 2 minutes to answer ;-) 

 I wish I would know more JS but i'm newbie in JS and i'm able to do small userscript but I will have to study more carefully to have a better understanding of JS to do more useful things.

Have a Great Week-End and Keep the good work !  :smiley:

**Comment by mikhoul** [comment]
There was a new TGM update today and I was thinking maybe... but same bug with ABP  :cry:

**Comment by tengwar** [comment]
@mikhoul Maybe it will work with another adblocker, like uBlock Origin?

**Comment by B00ze64** [comment]
Just as an FYI, the little drop down arrow that appears next to the Tab Groups button in Firefox is a list of tabs for the *current* group; it appears when there is no room for all the tabs in the Tab Bar. I've never seen a drop down to switch groups (and I run FF 38 right now, so I should have it if it existed, but it doesn't). This means that the little down-arrow should *still* be there for everyone, I don't think it was part of Panorama...

**Comment by Quicksaver** [comment]
@B00ze64 well, yeah... That's not what this thread is about though.

**Comment by B00ze64** [comment]
Lol, I know but twice now someone asked for the little arrow and you sent them here, Thought I'd explain that the arrow is not a Panorama thing here, since you closed those bugs too fast for me to reply there. Sorry if it's somewhat OT...

**Comment by Quicksaver** [comment]
Yeah but they asked for the arrow that selects groups and tabs in groups, not tabs, hence here. By the way, you can still comment in closed bugs if you think it's more appropriate there, for instance any case that you think really is about that open tabs arrow and not about a groups dropdown. ;)

**Comment by Quicksaver** [comment]
Added a quick access panel to [beta version 2b5](https://github.com/Quicksaver/Tab-Groups/releases/tag/v2b5). Please try it out and let me know what you think.

Note that beta has a lot of changes since the last release version, so it's definitely worth to give the how to use page a read-through.

For now the panel acts only as a quick switcher, to switch between groups and tabs without having to enter groups mode. Extra functionality **may** come later. ;)

**Comment by WolfganP** [comment]
Thanks @Quicksaver. Wanting to help on the beta testing, which "how to use beta" page are you specifically referring to?

**Comment by Quicksaver** [comment]
> which "how to use beta" page are you specifically referring to?

I should have quoted it. I meant the "How To Use" page in the add-on's preferences. ;)

**Comment by WolfganP** [comment]
Thanks @Quicksaver. Just for reference for others to read before installing, the guide is located at https://github.com/Quicksaver/Tab-Groups/blob/88137fcee57875a3be9fa44d179574fbd67fb463/chrome/locale/en-US/howto.dtd

**Comment by mikhoul** [comment]
It work fine for me BUT one thing I really don't like is to use the name of the Tab Group instead of the simple small button.

I use many group and sometime I use long name too so it take to many precious space on my toolbar that is already overloaded of useful addons ;-)

http://i.imgur.com/Obxy6DL.png

@Quicksaver 

Maybe just put back the Icon instead of the name since we know in which group we use so it's not really meaningful, at least for me. Alternatively having the option to use the name OR the button would be great too but more coding involved. :smiley_cat: 

I really like the menu to quick change the group directly from the toolbar, it will save me lot of time and it's so much intuitive too than changing screen.

Thanks :+1:

**Comment by Quicksaver** [comment]
> one thing I really don't like is to use the name of the Tab Group instead of the simple small button.

You can disable that in the preferences. ;)

**Comment by mikhoul** [comment]
Oups ! Overlooked  :smile:
